caffe2/ideep/operators/conv_op.cc

```diff
@@ -8,12 +8,17 @@ class IDEEPConvOp final : public IDEEPConvPoolOpBase {
   USE_IDEEP_CONV_POOL_BASE_FUNCTIONS();
 
   IDEEPConvOp(const OperatorDef& operator_def, Workspace* ws)
-      : IDEEPConvPoolOpBase(operator_def, ws),
-        training_mode_(
-            OperatorBase::GetSingleArgument<int>("training_mode", 0)) {
+      : IDEEPConvPoolOpBase(operator_def, ws) {
+    OPERATOR_NEEDS_FEATURE(order_ == StorageOrder::NCHW,
+        "Unsupported storage order.");
     OPERATOR_NEEDS_FEATURE(
         pad_l() == pad_r() && pad_t() == pad_b(),
         "Uneven padding not supported.");
+
+    attr_ = iattr();
+    algo_ = ialgo::convolution_direct;
+    training_mode_ = OperatorBase::GetSingleArgument<int>("training_mode", 0);
```

Comment:
Why don't we put this to the ctor initialization list as before?

Reply:
Trying to merge Conv and ConvFusion operators in order to removing redundant code, in #16309.
To remove compiling warning, move out of ctor init list.